### PR TITLE
Include additional properties in redis allocation status reports.

### PR DESCRIPTION
**Pull request by fancycode**
Additional properties:
- Socket type
- Local IP address / port
- Remote IP address / port
- SSL type
- SSH cipher

This helps with getting more information on TURN usage and to see where clients are connecting from.

### CODE REVIEW

**Comment by Neustradamus** [comment]
@misi: Happy New Year 2022!

Can you look the @fancycode PR?

Thanks in advance.

**Comment by ggarber** [comment]
I'm not sure if we should store the remote IP address...  There are some privacy concerns and legal issues with handling of Personal Identifiable Information that can make it problematic for some use cases.

Let me think about it.

**Comment by fancycode** [comment]
Thanks for looking into this!

Well, it's similar to a webserver log that contains remote addresses.

One use case is to find out that users from a given country (based on ip geolocation) are using TURN and eventually add servers closer to them to improve latency.

**Comment by ggarber** [comment]
> Well, it's similar to a webserver log that contains remote addresses.
> 
> One use case is to find out that users from a given country (based on ip geolocation) are using TURN and eventually add servers closer to them to improve latency.

I understand it, but at the same time there are use cases where you will want to not store IP addresses (look at #911 ) for example.

I think the solution is to make that configurable.   Maybe I can merge this change and then try to add that option.

**Review by ggarber — APPROVED** [review]

**Comment by ggarber** [comment]
@fancycode Can you do that small change?  I will merge it after that.  If you can't do it i will do it myself as soon as I can.  Thank you very much!

File: src/apps/relay/ns_ioalib_engine_impl.c [hunk 1]
```diff
@@ -3718,8 +3718,14 @@ void turn_report_allocation_set(void *a, turn_time_t lifetime, int refresh)
...
 					}
-					send_message_to_redis(e->rch, "set", key, "%s lifetime=%lu", status, (unsigned long)lifetime);
-					send_message_to_redis(e->rch, "publish", key, "%s lifetime=%lu", status, (unsigned long)lifetime);
+					uint8_t saddr[129];
+					uint8_t rsaddr[129];
+					addr_to_string(get_local_addr_from_ioa_socket(ss->client_socket), saddr);
+					addr_to_string(get_remote_addr_from_ioa_socket(ss->client_socket), rsaddr);
+					const char *ssl = ss->client_socket->ssl ? turn_get_ssl_method(ss->client_socket->ssl, "UNKNOWN") : "NONE";
+					const char *cipher = ss->client_socket->ssl ? get_ioa_socket_cipher(ss->client_socket) : "NONE";
+					send_message_to_redis(e->rch, "set", key, "%s lifetime=%lu, type=%s, local=%s, remote=%s, ssl=%s, cipher=%s", status, (unsigned long)lifetime, socket_type_name(get_ioa_socket_type(ss->client_socket)), saddr, rsaddr, ssl, cipher);
```
[the remark this answers is not in the document]
Comment (fancycode) [reply src/apps/relay/ns_ioalib_engine_impl.c]: Done

**Pull request merged**